crates/uv-python/fetch-download-metadata.py

```diff
@@ -540,6 +541,102 @@ async def _fetch_checksums(self, downloads: list[PythonDownload]) -> None:
             download.sha256 = checksums.get(download.filename)
 
 
+class GraalPyFinder(Finder):
+    implementation = ImplementationName.GRAALPY
+
+    RELEASE_URL = "https://api.github.com/repos/oracle/graalpython/releases"
+
+    PLATFORM_MAPPING = {
+        "windows": "windows",
+        "linux": "linux",
+        "macos": "darwin",
+    }
+
+    ARCH_MAPPING = {
+        "amd64": "x86_64",
+        "aarch64": "aarch64",
+    }
+
+    GRAALPY_VERSION_RE = re.compile(r"-(\d+\.\d+\.\d+)$", re.ASCII)
+    CPY_VERSION_RE = re.compile(r"Python (\d+\.\d+(\.\d+)?)", re.ASCII)
+    PLATFORM_RE = re.compile(r"(\w+)-(\w+)\.(?:zip|tar\.gz)$", re.ASCII)
+
+    def __init__(self, client: httpx.AsyncClient):
+        self.client = client
+
+    async def find(self) -> list[PythonDownload]:
+        downloads = await self._fetch_downloads()
+        await self._fetch_checksums(downloads, n=10)
+        return downloads
+
+    async def _fetch_downloads(self) -> list[PythonDownload]:
+        resp = await self.client.get(self.RELEASE_URL)
```

Comment:
By default, this pulls in just one page of releases, which is 30 items. Does it make sense to only keep the latest 30 releases for Graalpy?

Reply:
Yes, GraalPy releases 2 major and 4 patch releases a year, and our support window is only 2 years so we will support much more years back anyway, picking up 30 releases is plenty I'd say.

Reply:
Can you add an inline comment about that?

Reply:
Done